gammapy/cube/simulate.py

```diff
@@ -263,24 +264,31 @@ def run(self, dataset, observation=None):
         events : `~gammapy.data.EventList`
             Event list.
         """
-        events_src = self.sample_sources(dataset)
-
-        if dataset.psf:
-            events_src = self.sample_psf(dataset.psf, events_src)
-        else:
-            events_src.table["RA"] = events_src.table["RA_TRUE"]
-            events_src.table["DEC"] = events_src.table["DEC_TRUE"]
-
-        if dataset.edisp:
-            events_src = self.sample_edisp(dataset.edisp, events_src)
-        else:
-            events_src.table["ENERGY"] = events_src.table["ENERGY_TRUE"]
-
-        if dataset.background_model:
-            events_bkg = self.sample_background(dataset)
-            events = EventList.stack([events_bkg, events_src])
-        else:
-            events = events_src
+        try:
```

Comment:
When using the `try` `except` statement it is important to keep the code in the block as short as possible, otherwise all errors will be silently caught. If possible we should avoid it completely...

Can you change the implementation check e.g.
```
if len(dataset.models) > 1 and dataset.background_model is not None:
``` 
This should be simpler...

Reply:
Okay, thanks. I've removed it. 
However, I suggest that the condition should be `if len(dataset.models) > 1:` because, nested to that, there is already the condition `if dataset.background_model:`.